### PR TITLE
GetStockData and publish to topic

### PR DESCRIPTION
**Pull request by rishabhp58**
added functions to fetch stock data from an external api
added pubsub functions to pull messages from a topic and publish stock information to another topic

### CODE REVIEW

File: apis/model.go
Comment (vimalpatel19) [inline apis/model.go]: Can you change the name on this to **Ticker** and the json tag to **ticker** to be consistent. No need for **stock** to be pre-pended since the struct is named **StockData**.

Also, if you can just remove the firestore tag since this struct's data is currently not being stored in the firestore database.

**Comment by vimalpatel19** [comment]
Overall, looks good! You don't have to do this for this pull request but for future programming, try to think how you can incorporate the use of interface. The call to the external Stock API would have been a good use case for interface that way if we do ever switch to a different API, we just have to switch the implementation of the interface easily.

You should be good to go once you resolve the 2 merge conflicts. I'll be able to approve once that has been done.

**Review by vimalpatel19 — COMMENTED** [review]
For **StockData** struct, I meant to remove all firestore tags. And you can simply remove it rather than doing **firestore:"-"** so its cleaner that way.

**Comment by rishabhp58** [comment]
> For **StockData** struct, I meant to remove all firestore tags. And you can simply remove it rather than doing **firestore:"-"** so its cleaner that way.

Gotcha! Updated the pull request

**Review by vimalpatel19 — APPROVED** [review]

**Pull request merged**
